ci/repokitteh/modules/dco.star

```diff
@@ -0,0 +1,30 @@
+
+def pr_has_commits_without_dco(context, state):
+  return True
+
+def send_dco_message(sender):
+  lines = [
+    "hi @%s," % sender,
+    "  It seems that one or more of the commits in your Pull Request has not been signed.",
+    "  We require this to ensure we know and remember who made each contribution.",
+    ""
+    "  You can add the necessary commit hooks to sign your commits, by running the following in",
+    "  the root of the Envoy repository"
+    "  ```"
+    "  ./support/bootstrap"
+    "  git config --add alias.amend \"commit -s --amend\""
+    "  git config --add alias.c \"commit -s\""
+    "  ```"
+    ""
+    "  You will need to amend the problem commits to ensure they contain a line like:"
+    "  ```"
+    "  Signed-off-by: Joe Smith <joe@gmail.com>"
+    "  ```"
+    "  Full information can be found in the [contribution guidelines](https://github.com/envoyproxy/envoy/blob/master/CONTRIBUTING.md#dco-sign-your-work)"
+  github.issue_create_comment('\n'.join(lines))
+
+def _status(sender, context,  state):
+  if pr_has_commits_without_dco(context, state):
+    send_dco_message(sender)
+
+handlers.status(func=_status)
```

Comment:
@itayd i have tried to add a trigger for when the `DCO` bot reports back - which i assume  updates status

so far i havent got this working for 2 reasons
- this handler never seems to trigger
- i dumped the list of status events and it doesnt seem to include the event where the dco bot reports (it  seems to only have circleci events)

Reply:
im wondering if its possible to catch `check_run` events https://developer.github.com/webhooks/event-payloads/#check_run